client/state/mailchimp/settings/actions.js

```diff
@@ -0,0 +1,54 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+
+import {
+	MAILCHIMP_SETTINGS_LIST,
+	MAILCHIMP_SETTINGS_RECEIVE,
+	MAILCHIMP_SETTINGS_UPDATE,
+	MAILCHIMP_SETTINGS_UPDATE_SUCCESS,
+	MAILCHIMP_SETTINGS_UPDATE_FAILURE,
+} from 'state/action-types';
+import wpcom from 'lib/wp';
+
+export const requestSettings = siteId => ( {
+	siteId,
+	type: MAILCHIMP_SETTINGS_LIST,
+} );
+
+export function receiveSettings( siteId, lists ) {
+	return {
+		siteId,
+		type: MAILCHIMP_SETTINGS_RECEIVE,
+		lists,
+	};
+}
+
+export const requestSettingsUpdate = ( siteId, settings ) => {
+	return dispatch => {
+		dispatch( {
+			type: MAILCHIMP_SETTINGS_UPDATE,
+			siteId,
+			settings,
+		} );
+
+		return wpcom.req
+			.post( `/sites/${ siteId }/mailchimp/settings`, settings )
+			.then( () => {
+				dispatch( {
+					type: MAILCHIMP_SETTINGS_UPDATE_SUCCESS,
+					siteId,
+					settings,
+				} );
```

Comment:
The `POST` request doesn't return any response (or if it does, you ignore it) and `UPDATE_SUCCESS` action will just store exactly the same `settings` that the `UPDATE` action stored before sending the request. Is that useful?

It seems that this whole `POST` is just a fire-and-forget request where you don't do anything meaningful with the response. Then why write complicated `then` and `catch` handlers at all? :wink:

Reply:
Data layer handler for this `POST` request should actually work just fine -- I don't see any reason why it shouldn't.